enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/FileDataSecretEnricher.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.standard;
+
+import io.fabric8.kubernetes.api.builder.TypedVisitor;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.maven.core.util.Base64Util;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.MavenEnricherContext;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+public class FileDataSecretEnricher extends BaseEnricher {
+
+    protected static final String PREFIX_ANNOTATION = "maven.fabric8.io/secret/";
+
+    public FileDataSecretEnricher(MavenEnricherContext buildContext) {
+        super(buildContext, "fmp-secret-file");
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+        addAnnotations(builder);
+    }
+
+    private void addAnnotations(KubernetesListBuilder builder) {
+        builder.accept(new TypedVisitor<SecretBuilder>() {
+
+            @Override
+            public void visit(SecretBuilder element) {
+                final Map<String, String> annotations = element.buildMetadata().getAnnotations();
+                try {
+                    final Map<String, String> secretAnnotations = createSecretFromAnnotations(annotations);
+                    element.addToData(secretAnnotations);
+                } catch (IOException e) {
+                    throw new IllegalArgumentException(e);
+                }
+            }
+        });
+    }
+
+    private Map<String, String> createSecretFromAnnotations(final Map<String, String> annotations) throws IOException {
+        final Set<Map.Entry<String, String>> entries = annotations.entrySet();
+        final Map<String, String> secretFileLocations = new HashMap<>();
+
+        for(Iterator<Map.Entry<String, String>> it = entries.iterator(); it.hasNext(); ) {
```

Comment:
Hey @bohmber why not just use 

```
for (Map.Entry<String, String> entry : annotations.entrySet( ) )
```
?
The FMP code base uses this way of iteration everywhere, so it would look uniform with the codebase if you change it to this style.

Reply:
It's just a modified copy of the ConfigMapEnricher and i think the iterator is needed for the remove a few lines later 

`it.remove();`

Reply:
oh i see